File: frontend/src/types/material.ts
export type MaterialKind = "video" | "text" | "scientist";

export interface Material {
  id: string;
  kind: MaterialKind;
  kindLabel: string;
  title: string;
  body: string | null;
  deviceId: string | null;
  deviceName: string;
  clipKey: string | null;
  rfidUid: string | null;
  hasFile: boolean;
  originalName: string | null;
  mimeType: string | null;
  byteSize: number | null;
  createdBy: string;
  createdAt: string;
}

export const MATERIAL_KIND_OPTIONS: { id: MaterialKind; label: string; hint: string }[] = [
  { id: "video", label: "Ролик", hint: "Иллюминатор и голобоксы" },
  { id: "text", label: "Текст", hint: "Факт или подпись на экране" },
  { id: "scientist", label: "Учёный", hint: "Фигурка и метка RFID" },
];

File: backend/app/schemas/material.py
"""Публичные схемы материалов для пульта педагога."""

from pydantic import BaseModel


class MaterialPublic(BaseModel):
    """Карточка материала в списке и после сохранения."""

    id: str
    kind: str
    kindLabel: str
    title: str
    body: str | None = None
    deviceId: str | None = None
    deviceName: str
    clipKey: str | None = None
    rfidUid: str | None = None
    hasFile: bool
    originalName: str | None = None
    mimeType: str | None = None
    byteSize: int | None = None
    createdBy: str
    createdAt: str
